src/wms/src/main/java/org/geoserver/wms/featureinfo/FreemarkerStaticsAccessRule.java

```diff
@@ -0,0 +1,135 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms.featureinfo;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+import javax.lang.model.SourceVersion;
+import org.apache.log4j.Logger;
+
+/**
+ * Represents the rules for accessing static members from within Freemarker templates.
+ *
+ * @author awaterme
+ */
+class FreemarkerStaticsAccessRule {
+
+    public static final class RuleItem {
+        /** the class to allow access to */
+        private Class<?> clazz;
+        /** the alias (variable name) used to expose the statics of the class */
+        private String alias;
+        /**
+         * true, in case an number prefix was appended to keep the names distinct. Unlikely to occur
+         * in real world
+         */
+        private boolean numberedAlias;
+
+        public RuleItem(Class<?> clazz, String alias, boolean numberedAlias) {
+            super();
+            this.clazz = clazz;
+            this.alias = alias;
+            this.numberedAlias = numberedAlias;
+        }
+
+        public String getClassName() {
+            return clazz.getName();
+        }
+
+        public String getAlias() {
+            return alias;
+        }
+
+        public boolean isNumberedAlias() {
+            return numberedAlias;
+        }
+    }
+
+    /** Instance signals unrestricted access */
+    private static final FreemarkerStaticsAccessRule UNRESTRICTED =
```

Comment:
Having UNRESTRICTED and DISABLED be both using the default constructor is confusing. I see how it's working, checking an equality in Unrestricted, but it's quite odd. Please change it to have a constructor taking a boolean, the unrestricted flag, instead (so unrestricted would be ``new FreemarkerStaticsAccessRule(true)`` and disabled would be ``new FreemarkerStaticsAccessRule(false)``.

Reply:
Yes, you are right, I will adjust it.